src/main/java/uk/gov/hmcts/reform/sendletter/services/LetterService.java

```diff
@@ -38,13 +40,24 @@ public UUID send(LetterRequest letter, String serviceName) {
 
         log.info("Generated message: id = {} for letter with print queue id = {}", messageId, letter.type);
 
-        byte[] pdf = pdfCreator.create(letter);
-        Letter dbLetter = new Letter(messageId, serviceName, null, letter.type, pdf);
+        Optional<Letter> result = letterRepository.findByMessageId(messageId);
+
+        Letter dbLetter = result.filter(l -> l.getState().equals(LetterState.Created))
```

Comment:
It would be good if you log that the letter existed and hence returning existing letter id